src/util/prolog/prologResult.ts

```diff
@@ -17,11 +20,17 @@ export class PrologResult {
 		return this.results;
 	}
 
+	public getResultsFor(variable: string): string[] {
+		// Group 1 will be the content of the variable independent of the variable length.
+	 const pattern: any = new RegExp(variable + this.prologAnswerPattern, "g");
```

Comment:
Why does this need to be `any`?

Reply:
Seems like I misunderstood typescript types. It should rather be `RegExp` right?

Reply:
Yes, the type would be `RegExp`. Since you just assign `new RegExp` to this constant variable though, an explicit type is superfluous and doesn't increase typesafety nor readability. I'd just write `const pattern = new Regexp(...);`